src/axom/inlet/SphinxDocWriter.cpp

```diff
@@ -133,5 +114,65 @@ void SphinxDocWriter::writeAllTables() {
 
 }
 
+void SphinxDocWriter::collectFieldInfo(axom::sidre::Group* sidreGroup) {
+  std::vector<std::string> fieldAttributes(5, "");
+  fieldAttributes.resize(5);
+
+  fieldAttributes[0] = sidreGroup->getName();
+  if (sidreGroup->hasView("description")) {
+    fieldAttributes[1] = std::string(sidreGroup->getView("description")->getString());
+  } 
+
+  if (sidreGroup->hasView("defaultValue")) {
+    axom::sidre::TypeID type = sidreGroup->getView("defaultValue")->getTypeID();
+    if (type == axom::sidre::TypeID::INT8_ID) {
+      int8 val = sidreGroup->getView("defaultValue")->getData();

```

Comment:
Can this logic be cleaned up using a function that extracts the values and returns a sting?

E.g.  perhaps add the following to `inlet_utils`
```cpp
std::string getValueAsString(View* view)
{
   axom::sidre::TypeID type = view->getTypeID(); 
    if (type == axom::sidre::TypeID::INT8_ID) {
      int8 val = view->getData();
      return std::to_string(val);
    } else if (type == axom::sidre::TypeID::INT_ID) {
      int val = view->getData();
      return std::to_string(val);
    } else if (type == axom::sidre::DOUBLE_ID) {
      double val = view->getData();
      return std::to_string(val);
    } else {
      return view->getString();
    } 
}
```

You could then call it here as:
```cpp
   if (sidreGroup->hasView("defaultValue")) 
   { 
      fieldAttributes[2] = getValueAsString( sidreGroup->getView("defaultValue") );
   }
```

Similar calls throughout inlet (or at least this file) could clean up the code a lot.

Reply:
That's a very good idea, but note that INT8_ID is actually a boolean value in Inlet.  So that case would return "true" or "false"